example_test.go

```diff
@@ -48,18 +51,25 @@ func Example_sendAndReceive() {
 	fmt.Printf("Sent %v bytes\n", n)
 
 	// Read and print the response
+
 	buff := make([]byte, 100)
 	for {
 		// Reads up to 100 bytes
 		n, err := port.Read(buff)
 		if err != nil {
 			log.Fatal(err)
-			break
+
 		}
 		if n == 0 {
 			fmt.Println("\nEOF")
 			break
 		}
-		fmt.Printf("%v", string(buff[:n]))
+
+		if strings.Contains(string(buff[:n]), "\n\r") { //if we receive a newline and carriage return, stop reading
+
+			fmt.Printf("%v", string(buff)) //Print the data we received
+			break
+		}
+
```

Comment:
It may happen that the line arrives in multiple pieces, for example the string `hello\n`:
- may be read in a single call to `n, err := port.Read(buff)` that returns:
  `n == 6` and `buff == []byte("hello\n")`
or
- it may be read with two subsequent calls to read, for example:
  `n == 4` and `buff == []byte("hell")`
  `n == 2` and `buff == []byte("o\n")` 
it depends on how fast is the sender to transmit the whole line and how fast is the reader to process incoming data...

A better approach is to print everything from the serial port, and check only if you received an incoming `\n` (so we can handle different line endings linux:`\n` windows:`\n\r` mac:`\r\n`)

```suggestion
		fmt.Printf("%s", string(buff[:n]))

		// If we receive a newline stop reading
		if strings.Contains(string(buff[:n]), "\n") {
			break
		}
```

Reply:
okay thank you!